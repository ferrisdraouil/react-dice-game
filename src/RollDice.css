.RollDice {
  margin: 15px;
  display: inline-block;
}
