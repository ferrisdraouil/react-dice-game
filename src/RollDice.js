import React, { Component } from 'react';
import Die from './Die';
import './RollDice.css';

class RollDice extends Component {
  constructor(props) {
    super(props);
    this.state = {
      d1: 'one',
      d2: 'two'
    };
    this.handleClick = this.handleClick.bind(this);
  }

  roll() {
    const stateArr = ['one', 'two', 'three', 'four', 'five', 'six'];
    const randOne = stateArr[Math.floor(Math.random() * 6)];
    const randTwo = stateArr[Math.floor(Math.random() * 6)];
    this.setState({ d1: randOne, d2: randTwo });
  }

  handleClick(evt) {
    this.roll();
  }

  render() {
    return (
      <div>
        <div className="RollDice">
          <Die num={this.state.d1} />
          <Die num={this.state.d2} />
        </div>
        <div>
          <button onClick={this.handleClick}>Roll Dice</button>
        </div>
      </div>
    );
  }
}

export default RollDice;
