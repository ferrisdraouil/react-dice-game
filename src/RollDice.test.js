import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import RollDice from './RollDice';

it('renders without crashing', function() {
  shallow(<RollDice />);
});

it('matches snapshot', function() {
  let wrapper = shallow(<RollDice />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});
