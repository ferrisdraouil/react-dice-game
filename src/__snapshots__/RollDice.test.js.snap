// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    className="RollDice"
  >
    <Die
      num="one"
    />
    <Die
      num="two"
    />
  </div>
  <div>
    <button
      onClick={[Function]}
    >
      Roll Dice
    </button>
  </div>
</div>
`;
